File: crates/bts-map/tests/fixtures/greeter.go
// Fixture: Go greeter (bts-map snapshot test)
package main

import "fmt"

type Greeter struct{}

func (g Greeter) Greet(name string) string {
	return fmt.Sprintf("Hello, %s!", name)
}

func main() {
	g := Greeter{}
	fmt.Println(g.Greet("world"))
}

File: crates/bts-map/tests/fixtures/greeter.ts
// Fixture: TypeScript greeter (bts-map snapshot test)

interface Greeter {
    greet(name: string): string;
}

class SimpleGreeter implements Greeter {
    greet(name: string): string {
        return `Hello, ${name}!`;
    }
}

function main(): void {
    const g = new SimpleGreeter();
    const msg = g.greet("world");
    console.log(msg);
}
